ggml/src/ggml-backend.cpp

```diff
@@ -1394,6 +1394,12 @@ static enum ggml_status ggml_backend_sched_compute_splits(ggml_backend_sched_t s
                     }
                     ggml_backend_tensor_copy(input, input_cpy);
                 }
+                else {
+                    if (input_backend->iface.synchronize) {
+                        // async copy succeeded, need to synchronize the input backend to ensure the copy is done before the split backend uses it
+                        input_backend->iface.synchronize(input_backend);
+                    }
```

Comment:
A synchronization after an async copy is not necessary. The way async copy is intended to work is roughly explained here:
https://github.com/ggml-org/llama.cpp/blob/7fe03e7446ed4388539d3bcc013ae4e851b8d1e2/ggml/include/ggml-backend.h#L108-L112

Reply:
That makes sense, however i was receiving garbage output without this sync when using layer parallel. Is suspect it’s because the stream being used for the async copy is the source and not the dest, as specified in the code comments.

Reply:
I suppose you could remove the event wait on the dst stream at the end of the async copy, and transfer the responsibility of synchronizing the dst backend to the application.